docs/InputReferenceGuide.rst

```diff
@@ -0,0 +1,157 @@
+=====================
+Input Reference Guide
+=====================
+
+.. contents::
+   :local:
+
+Introduction
+============
+
+Reference guide to the LLVM-SPIRV-Translator input, which is compiler IR
+(intermediate representation) based on the `LLVM IR
+<https://llvm.org/docs/LangRef.html>`_. We will reference it as SPIR IR to
+not confuse it with vanilla LLVM IR.
+
+This document amends `SPIR-V representation in LLVM IR
+<https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/master/docs/SPIRVRepresentationInLLVM.rst>`_,
+which defines most of the rules for encoding SPIR-V instructions and types in
+LLVM IR.
+
+**TODO**: should we merge these two documents?
```

Comment:
I think it might make sense, because "SPIR-V representation in LLVM IR" documents mapping from LLVM IR to various SPIR-V features. For example, if you want to get a barrier or memory fence in SPIR-V, you need to read that document to see which LLVM IR should be generated